shark-graph/src/main/java/shark/ProguardMapping.kt

```diff
@@ -0,0 +1,120 @@
+package shark
+
+import java.io.File
+import java.io.FileNotFoundException
+import java.io.IOException
+import java.io.InputStream
+import java.text.ParseException
+
+
+class ProguardMapping private constructor(
```

Comment:
Not sure I understand. Do you mean an interface which current `ProguardMapping` class would be an implementation of? So that there can be other implementations not necessarily using files/input streams under the hood, is that correct?

Reply:
Exactly! You could either make ProguardMapping an interface and have e.g. `InputStreamProguardMapping` implementation, or keep `ProguardMapping` a simple class that holds the map and has the deobfuscate method, and then have a separate class that knows how to create a ProguardMapping from an InputStream.